chore: document stock vars in comments

jniles opened the pull request:
This PR adds comments to document in English on the stock lots calculations.  It also fixes a translation issue on the lots registry.

Closes #4721.

jniles committed:
docs(stock): add documenting comments to stock code

Adds comments so I don't forget what these variables are.

jniles committed:
fix(lots): duration renders in the correct lang

Makes the i18n for the duration of stock lots render correctly.

Closes #4721.

jmcameron commented on server/controllers/stock/core.js:
```diff
@@ -402,14 +408,38 @@ function stockManagementProcess(inventories) {
     Q = inventory.quantity; // the quantity
     CM = inventory.avg_consumption; // consommation mensuelle
     CM_NOT_ZERO = !CM ? 1 : CM;
+
```
Thanks for adding these comments!   I was looking at these lines yesterday and thought that we should beef up the comments!   Great addition!

mbayopanda commented on server/controllers/stock/core.js:
```diff
@@ -402,14 +408,38 @@ function stockManagementProcess(inventories) {
     Q = inventory.quantity; // the quantity
```
We can also mention that this quantity is the available quantity of inventory for a given depot, the implementation is in : `/server/controllers/stock/core.js`

jniles replied:
Great idea.  Thank you!

mbayopanda commented on server/controllers/stock/core.js:
```diff
@@ -402,14 +408,38 @@ function stockManagementProcess(inventories) {
     Q = inventory.quantity; // the quantity
     CM = inventory.avg_consumption; // consommation mensuelle
```
The `avg_consumption` is the CMM or AMC which is calculated during stock consumption (stock exit to patient or to service, or to a depot if the depot is a warehouse), implementation is in `/server/controllers/stock/index.js`

mbayopanda reviewed (approved):
LGTM 👍 

I just add some comment about the quantity and the avg_consumption.

jmcameron commented on client/src/modules/stock/lots/registry.js:
```diff
@@ -133,6 +134,12 @@ function StockLotsController(
+          const delay = moment(new Date(lot.expiration_date)).diff(current);
+          lot.delay_expiration = moment.duration(delay).humanize();
+        });
 
```
Somehow this is losing the sign of the expiration when the lot is expired.  For expired items, in English, we should append "ago", and in French prepend "il y a"  (or something similar).  Or instead saying 'un mois' leave it as a digit and prepend a negative sign?  See below.

jmcameron replied:
![image](https://user-images.githubusercontent.com/62145/90416810-cf284e80-e067-11ea-8a20-5207fe3d3842.png)

jmcameron replied:
The translation of 'year' / 'ans', 'month' / 'mois' looks good.

jniles replied:
I've added a direction ("in two months") and ("two months ago").  

![image](https://user-images.githubusercontent.com/896472/90486451-dc5d4000-e130-11ea-93d5-3c0df43006f2.png)

What is your view on that presentation?

Also, I noticed that we have some bugs in one of our production databases where expired stock isn't showing up in red.  I'm not sure why that is the case, but I'll have to investigate.

jmcameron commented on server/controllers/stock/core.js:
```diff
@@ -389,27 +389,62 @@ function getLotsOrigins(depotUuid, params) {
+    // Compute Minimum Stock
+    // The minumum of stock required is double the security stock.
     inventory.S_MIN = inventory.S_SEC * 2; // stock minimum
+
```
We should probably make that '2' a parameter.  And should it be possible to assign different values to different inventory items?

jmcameron commented on server/controllers/stock/core.js:
```diff
@@ -389,27 +389,62 @@ function getLotsOrigins(depotUuid, params) {
+    // compute the inventory status code
     if (Q <= 0) {
       inventory.status = 'stock_out';
       inventory.stock_out_date = inventory.last_movement_date;
```
I see that this section has been expanded in the current code to include other status values ('security_reached', 'minimum_reached', etc).   I wonder if these should be Bhima constants somewhere?

jniles replied:
Yeah, this is probably true.  I'll make an issue.

jmcameron commented on server/controllers/stock/core.js:
```diff
@@ -389,27 +389,62 @@ function getLotsOrigins(depotUuid, params) {
+
+    // Compute the Risk of Stock Expiry
+    // The risk of stock expiry is the quantity of drugs that will expire before you
+    // can use them.
```
Thanks for adding the comments!

jmcameron reviewed (approved):
Once you fix the missing negative on the expiration of expired lots, this is ready to go.  The other comments are minor or could be handled  in a later issue.

jniles committed:
chore(stock): directionality in moment.humanize()

Adds directionality to the stock lots expiration dates to ensure that we
can read them without confusion.  For example "two months ago" rather
than just "two months".

jmcameron reviewed (approved):
It looks great!    Don't forget to add an issue for making 'stock_out', etc, into BHIMA constants (https://github.com/IMA-WorldHealth/bhima/blob/master/server/controllers/stock/core.js#L414).
